src/System.Private.ServiceModel/src/System/ServiceModel/Channels/MessageContent.cs

```diff
@@ -180,15 +179,34 @@ protected override Task<Stream> CreateContentReadStreamAsync()
             // WriteMessageAsync might run synchronously and try to write to the stream. ProducerConsumerStream
             // will block on the write until the stream is being read from. The WriteMessageAsync method needs
             // to run on a different thread to prevent a deadlock.
-            _stream = new ProducerConsumerStream();
-            var bufferedStream = new BufferedWriteStream(_stream, WriteBufferSize);
-            Task.Run(() => _messageEncoder.WriteMessageAsync(_message, bufferedStream));
-            return Task.FromResult(_stream);
+            var resultStream = new ProducerConsumerStream();
```

Comment:
Disposing a BufferedWriteStream also disposes the wrapped stream. But it also does one other step when Dispose is called, it flushes out any unwritten buffered data to the inner stream. Our message encoders will already do this, but if an external MessageEncoder didn't call Flush, then we might not send the final bytes of a request. This was a stress bug waiting to be reported :)